src/parser/mod.rs

```diff
@@ -7379,6 +7380,31 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse an REPLACE statement
+    pub fn parse_replace(&mut self) -> Result<Statement, ParserError> {
+        if !dialect_of!(self is MySqlDialect) {
+            return parser_err!("unmatched dialect", self.peek_token().location);
```

Comment:
If a user gets a "unmatched dialect" error when parsing their SQL I think it will be very confusing and will not tell them how to fix the problem

Can you please change the error to say something like "Unsupported statement REPLACE" or "unexpected keyword REPLACE"?

Reply:
Fixed error message